### ISSUE TITLE
[$20]"Topcoder handle" name foreground and background colors - consistent change across all pages per WCAG 2 AA contrast ratio 

### ISSUE THREAD

**Issue by lakshmiathreya** [topcoder-platform/community-app#2996]
**Describe the bug**
Refer [MSFT-228] 

**To Reproduce**
Refer [MSFT-228] - verify on across all pages on Community and Accounts pages.

**Expected behavior**
Consistent foreground and background colors across all pages for Topcoder handle.

**Desktop (please complete the following information):**
 - OS: Any
 - Browser: Any

**Smartphone (please complete the following information):**
 - Device: Any
 - OS: Any
 - Browser: Any
 

**Additional context**

**Comment by lakshmiathreya** [comment]
Fyi - @nithyaasworld @drasticdpk @tosha5252

**Comment by Oanh-and-only-Oanh** [comment]
@veshu, please assign prize amount and open for tcx pickup.

**Comment by veshu** [comment]
@Oanh-and-only-Oanh I remember there was similar issue already worked. I will check that and update this issue accordingly.

@codeMinter Please help me to find the issue which is similar. Thanks!

**Comment by veshu** [comment]
Ok found the similar issue https://github.com/topcoder-platform/community-app/issues/2886, I hoped that issue will covered all 5 rating band. 

Since we have only covered 1 (gray rated member) we need to update for 
- green
- gold
- red
- black

@Dara-K Please provide the color for this bands. Thanks!

**Comment by PrakashDurlabhji** [comment]
@veshu it was with me, this is issue you are talking about https://github.com/topcoder-platform/community-app/issues/2886.

Am I correct? then can i get this issue?

**Comment by codeMinter** [comment]
Sorry @veshu just saw but looks like you found one. 👍

**Comment by PrakashDurlabhji** [comment]
@veshu we both commented on exact same time. what a timing

**Comment by veshu** [comment]
@PrakashDurlabhji Yes you can pick as long as you don't have pending issues/PRs. Thanks!

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30096709 has been created for this ticket.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30096709 has been updated - the new changes has been updated for this ticket.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30096709 has been updated - it has been assigned to rashmi73.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by Dara-K** [comment]
@veshu  Here are the other colors for the handles:

•	Gray: #555555
•	Pastel-Blue: #5656F4
•	Green: #328732
•	Gold: #9A6D00
•	Red: #EA1900
•	Black: #151516

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30096713 has been created for this ticket.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30096713 has been updated - it has been assigned to rashmi73.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by veshu** [comment]
@rashmi73 Please check the color and submit the fix as soon as possible, we need to move faster. Thanks!

**Comment by rashmi73** [comment]
@veshu yes sure in 10 minutes

**Comment by rashmi73** [comment]
@veshu https://github.com/topcoder-platform/community-app/pull/3058 ready

**Comment by codeMinter** [comment]
@rashmi73 your PR is failing. Please don't flag for Review unless everything is clean on PRs.

**Comment by rashmi73** [comment]
@codeMinter raised new PR https://github.com/topcoder-platform/community-app/pull/3061

**Comment by rashmi73** [comment]
@codeMinter @veshu PR ready https://github.com/topcoder-platform/community-app/pull/3061

**Comment by codeMinter** [comment]
> @veshu Here are the other colors for the handles:
> 
> •	Gray: #555555
> •	Pastel-Blue: #5656F4
> •	Green: #258025
> •	Gold: #9A6D00
> •	Red: #EA1900
> •	Black: #151516

@Dara-K I see we have PR for this issue from @rashmi73 based on color suggested, but only passing are RED and Black AA level. Can you help get AA level passing color codes ?

**Comment by veshu** [comment]
@Dara-K Can you please help us here, Thanks!

**Comment by Dara-K** [comment]
Hi @veshu @codeMinter ,
When I set those color codes, I tested them to pass the AA test and they show up well for me. Did I miss anything in its testing?

Here's a short video how I did it (tested all the 4 above colors: grays,purples. greens, gold):
https://drive.google.com/file/d/1O6hEtFxEWG30rDSqGG9nuTcO_D8jKN4g/view?usp=sharing

**Comment by veshu** [comment]
Thanks @Dara-K your provided code looks good. 

@codeMinter @rashmi73 The issue is current fix is fixing the data on list of registrants as it looks coming from API 

https://monosnap.com/file/LTLjN1IPvwYO8n97TffPWsyvIhdnLi

@sushilshinde @Oanh-and-only-Oanh @crazyk07 Please guide here where to fix this issue which needs to be one API.

**Comment by cwdcwd** [comment]
Those values are hard coded in the challenge API [here](https://github.com/appirio-tech/ap-challenge-microservice/blob/dev/src/main/java/com/appirio/service/challenge/Helper.java#L101)

**Comment by skyhit** [comment]
@cwdcwd these are the color style from the legacy system, so we are going to change it?

also here - https://github.com/topcoder-platform/tc-elasticsearch-feeder-service/blob/423d70e4f31b1fef1117d8da28f90489df8e5ba7/src/main/java/com/appirio/service/challengefeeder/manager/ChallengeFeederUtil.java#L84

probably, we should return rating, and let the frontend determine that?

**Comment by sushilshinde** [comment]
@skyhit for now let's only change color codes

**Comment by sushilshinde** [comment]
@skyhit @veshu @Dara-K here is the change and few things are missing from the requirements 

```
if (rating == null) {
    return "color: #151516"; // CHANG
}

if (rating < 0) {
    return "color: #FF9900"; // orang
}
if (rating < 900) {
    return "color: #555555";// gray C
}
if (rating < 1200) {
    return "color: #258025";// green 
}
if (rating < 1500) {
    return "color: #5656F4";// blue C
}
if (rating < 2200) {
    return "color: #DDCC00";// yellow
}
if (rating > 2199) {
    return "color: #EA1900";// red CH
}
// return black otherwise.
return "color: #151516"; // CHANGED
```

**Comment by veshu** [comment]
@sushilshinde @skyhit 
yellow is Gold(9A6D00)
orange (BA4C00)

**Comment by sushilshinde** [comment]
```
	if (rating == null) {
            return "color: #151516"; // CHANGED
        }

        if (rating < 0) {
            return "color: #BA4C00"; // orange CHANGED
        }
        if (rating < 900) {
            return "color: #555555";// gray CHANGED
        }
        if (rating < 1200) {
            return "color: #258025";// green CHANGED
        }
        if (rating < 1500) {
            return "color: #5656F4";// blue CHANGED
        }
        if (rating < 2200) {
            return "color: #9A6D00";// yellow CHANGED
        }
        if (rating > 2199) {
            return "color: #EA1900";// red CHANGED
        }
        // return black otherwise.
        return "color: #151516"; // CHANGED
```
@veshu please review the change

**Comment by veshu** [comment]
@sushilshinde looks good. Thanks!

**Comment by rashmi73** [comment]
@veshu @sushilshinde shall i update PR make changes above?

and yes yellow is gold, I assumed it in my PR

**Comment by veshu** [comment]
@rashmi73 yes please update accordingly. Thanks!

**Comment by codeMinter** [comment]
@sushilshinde @veshu we need change in PR as has conflicts, see it Is marked `Ready for QA`.

**Comment by rashmi73** [comment]
@codeMinter I hope my change is correct. PR update
@sushilshinde 
@veshu

**Comment by lakshmiathreya** [comment]
Verified consistency in colors. However:
-- WAVE continues to show significant number of contrast issues. This is a generic Issue raised by me to review all colors and all contrasts across pages. Would like to review collectively at a later date. 
-- The name 'topcoder' which is part of the logo - does not show up under contrast review - however, request a review for visibility since the contrast appears low with the current shade of black.

**Comment by tosha5252** [comment]
@veshu this ticket has been satisfied; however, there are other contrast issues with the page...note comment above by @lakshmiathreya...can another issue be created before this one is closed to be sure those are addressed?

**Comment by veshu** [comment]
@tosha5252 @lakshmiathreya separated the overall verification to separate issue. #3199

**Comment by tosha5252** [comment]
changes verified

Windows 10 env
Browser: Chrome

**Comment by crazyk07** [comment]
Payment task has been updated: https://software.topcoder.com/review/actions/ViewProjectDetails?pid=30096713<br/><br/>```This is an automated message for crazyk via Topcoder X```
